tests/unit/test_app.py

```diff
@@ -1816,6 +1816,17 @@ def test_http_response_to_dict(body, headers, status_code):
     assert isinstance(serialized['body'], six.string_types)
 
 
+@given(headers=STR_MAP,
```

Comment:
Not sure we need these property based tests for this change.  We're passing in varying headers and status code values but our test is just verifying the values of sets.  You can probably just create various `Response` instances and verify the `to_dict()` serializes the types properly.  Something like `test_can_encode_binary_body_as_base64()`. 

Probably worth adding an extra test for nested sets as well.